apps/server/lib/lexical/server/transport/std_io.ex

```diff
@@ -59,11 +61,16 @@ defmodule Lexical.Server.Transport.StdIO do
       "\n" ->
         headers = parse_headers(buffer)
 
-        with {:ok, content_length} <-
-               header_value(headers, "content-length", &String.to_integer/1),
+        with {:ok, content_length} <- content_length(headers),
              {:ok, data} <- read_body(device, content_length),
              {:ok, message} <- JsonRpc.decode(data) do
           callback.(message)
+        else
+          {:error, :empty_response} ->
+            nil
+
+          {:error, reason} ->
+            Logger.critical("error reading protocol message: #{inspect(reason)}")
```

Comment:
The reason will likely be less helpful than logging some or all of the data.

Reply:
I think this log message's worthwhile for now, better than nothing.